frontend/src/store.js

```diff
@@ -2,13 +2,20 @@ import {createStore, combineReducers, applyMiddleware} from 'redux'
 import thunk from 'redux-thunk'
 import {composeWithDevTools} from 'redux-devtools-extension'
 import {productDetailsReducer, productListReducer} from './reducers/productReducers'
+import {cartReducer} from './reducers/cartReducers'
 
 const reducer = combineReducers({
     productList: productListReducer,
-    productDetails: productDetailsReducer
+    productDetails: productDetailsReducer,
+    cart:cartReducer
 })
 
-const initialState = {}
+const cartItemsFromStorage = localStorage.getItem('cartItems') ?
```

Comment:
Try not to access local storage too many times as much as you can
`const stringCartItemsFromStorage = localStorage.getItem('cartItems')`
`const cartItemsFromStorage = stringCartItemsFromStorage ? JSON.parse(stringCartItemsFromStorage) : []`

So here we accessed local storage only once expect of twice

Reply:
@MoatasemEbrahim why we should avoid accessing the localStorage many times? i did not get your point

Reply:
Is it related with performance or runtime issues ?